server/test/run.go

```diff
@@ -159,9 +165,44 @@ func (r Run) SuccessfulLinterExecution(linter analyzer.LinterResult) Run {
 	r.State = RunStateAwaitingTestResults
 	r.Linter = linter
 
+	if !r.Linter.Passed {
+		r = r.RequiredGateFailed(testrunner.RequiredGateAnalyzerRules)
+	}
+
+	if r.Linter.Score < linter.MinimumScore {
+		r = r.RequiredGateFailed(testrunner.RequiredGateAnalyzerScore)
+	}
+
 	return r
 }
 
+func (r Run) ConfigureRequiredGates(gates []testrunner.RequiredGate) Run {
```

Comment:
This logic is to support side effects from the linter and analyzer results to update the gates.
It also includes logic to generate the results from a test run without it for retro-compat